README.md

```diff
@@ -94,31 +94,56 @@ Once you have implemented the tools, fill out the report below.
+During the testing we realized that the alphabets used when generating reads and genomes had a big impact on the performance of the naive algorithm. This boils to the fact that the algorithm quickly decides that the current placement of the read is not a match in the genome it is searching. This means that the naive algorithm behaves as a linear algorithm on these kinds of input. Thee naive algorithm therefore tends to outperform KMP since KMP needs to pay the overhead of creating the borderarray. After choosing an alphabet with less possible characters we made the naive algorithm actually increase quadratically as is also shown later in the report.
+
+We also came to the conclusion that making weekly exercises was a good idea since a large part of the project was already done.
+
 
 ### Problems encountered if any. 
+During implementation we found that the default buffersize of bufio.NewScanner is 2^16 bytes which caused problems during testing since when we wanted to test runningtime on genomes larger than 65536 characters the parser returned an empty string.
+
+In the beginning we also faced some problems with the folder structure in regards to using our implementations of naive and lin in testing and thus we resorted to creating a stub of both in the algos_test.go file to then use when testing.
 
-*Describe this here.*
 
 ### Experiments that verifies the correctness of your implementations.
 
-*Describe this here.*
+We verified the implementation of KMP by comparing the result to the result of the naive algorithm. We ran both algorithms on randomly generated data over different alphabets:  DNA, English alphabet, A, AB and (AA...AAB) and alphabets where occourences of some letters were way more common than others, etc. We also tried some edgecases such as empty x or p strings. In all cases the two algorithms generated reported the same matches. So we concluded that since both algorithms yielded some matches and the reads and genomes were sufficiently long and random, our implementation of KMP works.
+
 
 ### Experiments validating the running time.
 
 For this section, you should address the following:
 
 * An experiment that verifies that your implementation of `naive` uses no more time than O(nm) to find all occurrences of a given pattern in a text. Remember to explain your choice of test data. What are “best” and “worst” case inputs? 
 
+To show the `naive` algorithm runs in O(nm) we conducted tests with a variable x value and different fixed p values.
+These results can be seen in:
+
+![](figs/naive_p_constant.jpg)
+
+In this experiment we run the naive algorithm on a repetitive alphabet which is prone to containing a lot of matches before a mismatch occurs. This results in an expected worst case O(nm) performance.
+It can clearly be seen that an increase in p value from 100 to 200 results in linear growth in the running time. This can be seen since the running time roughly doubles between the two runs of the algorithm.
```

Comment:
However, is that what you expect from $O(nm)$? Or is it the expected behaviour from $O(n+m)$?

Let's consider $f(n,m) = 5n + m \in O(n+m)$. If I plot $n\mapsto f(n,1) = 5n + 1$ and $n\mapsto f(n,2) = 5n+2$ I double the intercept (but the slope doesn't change). To me, your figure looks like it is the intercept changing and not the slope.

If we have $f(n,m) = 5 n m \in O(nm)$, plotting $n\mapsto f(n,1) = 5n$ and $n\mapsto f(n,2) = 10n$ I've doubled the slope (and not the intercept in this case, but you can add a term for an isolated $m$ if you want that).

Doesn't the graph look more like a constant slope and changing intercept rather than a constant intercept but changing slope?